tests/zfs-tests/tests/functional/cp_files/cp_stress.ksh

```diff
@@ -0,0 +1,69 @@
+#! /bin/ksh -p
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or https://opensource.org/licenses/CDDL-1.0.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+
+#
+# Copyright (c) 2023 by Lawrence Livermore National Security, LLC.
+#
+
+. $STF_SUITE/include/libtest.shlib
+
+#
+# DESCRIPTION:
+# This test is for the dirty dnode SEEK_HOLE/SEEK_DATA bug described in
+# https://github.com/openzfs/zfs/issues/15526
+#
+# The bug was fixed in https://github.com/openzfs/zfs/pull/15571 and
+# was backported to 2.2.2 and 2.1.14.  This test case is just to
+# make sure it does not come back.
+#
+# STRATEGY:
+#
+# 1. Run the 'seekflood' binary, for creation of files with timing
+#    characteristics that can trigger #15526.
+# 2. A single run is not always a trigger, so run repeatedly.
+
+verify_runnable "global"
+
+function cleanup
+{
+	rm -rf /$TESTDIR/cp_stress
+}
+
+log_assert "Run the 'seekflood' binary repeatedly to try to trigger #15526"
+
+log_onexit cleanup
+
+log_must mkdir /$TESTPOOL/cp_stress
+
+MYPWD="$PWD"
+cd /$TESTPOOL/cp_stress
+CPUS=$(get_num_cpus)
+
+# Results from buildbot suggest that fewer than 17 runs may be insufficient to
+# trigger failure.
+#
+# Cautiously perform 25 runs.  Each run takes around 12 seconds.
```

Comment:
There's really no right number for the minimum number of iterations needed to hit this reliably.  In practice we can rely on the fact that we run this test frequently and any issue will be frequently observed in the CI.  I'd suggest we drop this to 10 runs to keep this to about 2 minutes of run time per run of the test suite.

Reply:
Ok I'll drop it to 10